packages/nextjs/src/server/utils.ts

```diff
@@ -182,7 +182,7 @@ export function getRequestUrl({ request, relativePath }: { request: Request; rel
 
   // X-Forwarded-Proto could be 'https, http'
   const protocol =
-    (headers.get('X-Forwarded-Proto') ?? (headers as any)['X-Forwarded-Proto'])?.split(',')[0] ?? url.protocol;
+    (headers.get('X-Forwarded-Proto') ?? (headers as any)['X-Forwarded-Proto'])?.split(',')[0] ?? (url.protocol && url.protocol.slice(0, -1));
```

Comment:
This feels more robust, thanks for raising this 😀
```suggestion
    (headers.get('X-Forwarded-Proto') ?? (headers as any)['X-Forwarded-Proto'])?.split(',')[0] ?? (url.protocol && url.protocol.replace(/\:$/, '');
```